aws/sra-test/integration-tests/aws-sdk-s3/tests/interceptors.rs

```diff
@@ -0,0 +1,120 @@
+    };
+
+    let resp = dbg!(
+        client
+            .list_objects_v2()
+            .config_override(
+                aws_sdk_s3::Config::builder().interceptor(util::TestUserAgentInterceptor)
+            )
+            .bucket("test-bucket")
+            .prefix("prefix~")
+            .send_orchestrator_with_plugin(Some(fixup))
+            .await
+    );
+    let resp = resp.expect("valid e2e test");
+    assert_eq!(resp.name(), Some("test-bucket"));
+    conn.full_validate(MediaType::Xml).await.expect("failed")
+}
+
+#[derive(Debug)]
+struct RequestTimeResetInterceptor;
+impl Interceptor for RequestTimeResetInterceptor {
+    fn modify_before_signing(
+        &self,
+        _context: &mut InterceptorContext<BeforeTransmit>,
+        cfg: &mut ConfigBag,
+    ) -> Result<(), aws_smithy_runtime_api::client::interceptors::BoxError> {
+        cfg.set_request_time(RequestTime::new(UNIX_EPOCH));
+
+        Ok(())
+    }
+}
+
+#[derive(Debug)]
+struct RequestTimeAdvanceInterceptor(Duration);
+impl Interceptor for RequestTimeAdvanceInterceptor {
+    fn modify_before_signing(
+        &self,
+        _context: &mut InterceptorContext<BeforeTransmit>,
+        cfg: &mut ConfigBag,
+    ) -> Result<(), aws_smithy_runtime_api::client::interceptors::BoxError> {
+        let request_time = cfg.request_time().unwrap();
+        let request_time = RequestTime::new(request_time.system_time() + self.0);
+        cfg.set_request_time(request_time);
+
+        Ok(())
+    }
+}
+
+#[tokio::test]
+async fn interceptor_priority() {
+    let conn = dvr::ReplayingConnection::from_file(LIST_BUCKETS_PATH).unwrap();
+
+    // `RequestTimeResetInterceptor` will reset a `RequestTime` to `UNIX_EPOCH`, whose previous
+    // value should be `SystemTime::now()` set by `FixupPlugin`.
+    let config = aws_sdk_s3::Config::builder()
+        .credentials_provider(Credentials::for_tests())
+        .region(Region::new("us-east-1"))
+        .http_connector(DynConnector::new(conn.clone()))
+        .interceptor(util::TestUserAgentInterceptor)
+        .interceptor(RequestTimeResetInterceptor)
+        .build();
+    let client = Client::from_conf(config);
+    let fixup = util::FixupPlugin {
+        timestamp: SystemTime::now(),
+    };
+
+    // `RequestTimeAdvanceInterceptor` configured at the operation level should run after,
+    // expecting the `RequestTime` to move forward by the specified amount since `UNIX_EPOCH`.
+    let resp = dbg!(
+        client
+            .list_objects_v2()
+            .config_override(aws_sdk_s3::Config::builder().interceptor(
+                RequestTimeAdvanceInterceptor(Duration::from_secs(1624036048))
+            ))
+            .bucket("test-bucket")
+            .prefix("prefix~")
+            .send_orchestrator_with_plugin(Some(fixup))
+            .await
+    );
+    let resp = resp.expect("valid e2e test");
+    assert_eq!(resp.name(), Some("test-bucket"));
+    conn.full_validate(MediaType::Xml).await.expect("failed")
```

Comment:
`expect("failed")`? I think this is flipped. Should be `expect("success")`.

Reply:
Ah, copied & pasted without realizing it. I've seen two camps for what to say in `.expect`: one camp for saying what should've happened and the other for saying what actually happened. I've been arbitrarily choosing one over the other whenever I use `.expect`, but saying what should've happened makes more sense to me now. Addressed in 44af0db.

Reply:
I take the stance of reading it as plain English: "do this action and _expect_ success"